src/detect-engine-analyzer.c

```diff
@@ -51,6 +51,69 @@ typedef struct FpPatternStats_ {
     uint64_t tot;
 } FpPatternStats;
 
+/* Details for each buffer being tracked */
+typedef struct DetectEngineAnalyzerItems {
+    uint32_t    bufcount;
+    int16_t     item_id;
+    bool        requires_external_bufcount;
```

Comment:
Again, perhaps a better name for this could be chosen. `External` means that the item is explicitly checked in a stand alone test case as shown here: "External" means the item is being tracked with the (poorly named `ExternalBufCount` struct) separately -- both `http_method_buf` and `http_server_body_buf` are "external".

```
    if (http_method_buf) {
        if (rule_flow && rule_flow_toclient) {
            rule_warning += 1;
            warn_method_toclient = 1;
        }
        if (http_server_body_buf) {
            rule_warning += 1;
            warn_method_serverbody = 1;
        }
        if (rule_content == 0 && rule_content_http == 0 && (rule_pcre > 0 || rule_pcre_http > 0)) {
            rule_warning += 1;
            warn_pcre_method = 1;
        }
    }
```